ui/pom.xml

```diff
@@ -152,4 +146,63 @@
       </plugin>
     </plugins>
   </build>
+  
+  <profiles>
+    <profile>
+      <id>build-docker-image</id>
+      <activation>
+        <property>
+          <name>env.DOCKER_HOST</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <version>1.8</version>
+            <executions>
+              <execution>
+                <phase>package</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+              </execution>
+            </executions>
+            <configuration>
+              <target>
+                <sync todir="${basedir}/docker/dist">
+                  <fileset dir="${basedir}/dist"/>
+                </sync>
+              </target>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <version>${docker.maven.plugin.fabric8.version}</version>
```

Comment:
We could also use the fabric8-maven-plugin here for doing the build so we can still work with image-streams on minishift (and automatic redeployments), also for the ui.

The advantage using stream should be that the pods get automatically redeployed after the build without the need to explicitly kill them. 

I'm happy to merge now, and switch over later to fabric8-maven-plugin.

Reply:
Yeah that sounds like a good idea.